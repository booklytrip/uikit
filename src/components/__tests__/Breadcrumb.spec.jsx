import React from 'react';
import { mount } from 'enzyme';
import renderer from 'react-test-renderer';

import {
    Breadcrumb,
    BreadcrumbSection,
    BreadcrumbDivider,
} from '../Breadcrumb';

describe('Breadcrumb', () => {
    it('renders without crashing', () => {
        mount(
            <Breadcrumb>
                <BreadcrumbSection>Item 1</BreadcrumbSection>
                <BreadcrumbDivider />
                <BreadcrumbSection>Item 2</BreadcrumbSection>
            </Breadcrumb>
        );
    });
    
    it('should call onClick event if clicked on section', () => {
        const spy = jest.fn();
        const section = mount(<BreadcrumbSection onClick={ spy } />);
        section.simulate('click');
        
        expect(spy).toHaveBeenCalled();
    });
    
    describe('Snapshots', () => {
        it('renders in different size', () => {
            const rendered = renderer.create(
                <Breadcrumb size="large">
                    <BreadcrumbSection>Item 1</BreadcrumbSection>
                    <BreadcrumbDivider />
                    <BreadcrumbSection>Item 2</BreadcrumbSection>
                </Breadcrumb>
            );
            
            expect(rendered.toJSON()).toMatchSnapshot();
        });
        
        it('renders section as a link', () => {
            const rendered = renderer.create(
                <BreadcrumbSection link>Item 1</BreadcrumbSection>
            );
            
            expect(rendered.toJSON()).toMatchSnapshot();
        });

        it('renders active section', () => {
            const rendered = renderer.create(
                <BreadcrumbSection active>Item 1</BreadcrumbSection>
            );
            
            expect(rendered.toJSON()).toMatchSnapshot();
        });
        
        it('renders divder with an icon', () => {
            const rendered = renderer.create(
                <BreadcrumbDivider icon="right arrow" />
            );
            
            expect(rendered.toJSON()).toMatchSnapshot();
        });
    });
});
