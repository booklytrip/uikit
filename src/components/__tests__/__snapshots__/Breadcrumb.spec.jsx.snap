exports[`Breadcrumb Snapshots renders active section 1`] = `
<div
  className="section active"
  href={undefined}
  onClick={undefined}>
  Item 1
</div>
`;

exports[`Breadcrumb Snapshots renders divder with an icon 1`] = `
<i
  className="divider right arrow icon" />
`;

exports[`Breadcrumb Snapshots renders in different size 1`] = `
<div
  className="ui breadcrumb large">
  <div
    className="section"
    href={undefined}
    onClick={undefined}>
    Item 1
  </div>
  <div
    className="divider">
    /
  </div>
  <div
    className="section"
    href={undefined}
    onClick={undefined}>
    Item 2
  </div>
</div>
`;

exports[`Breadcrumb Snapshots renders section as a link 1`] = `
<a
  className="section"
  href={undefined}
  onClick={[Function]}>
  Item 1
</a>
`;
